lib/checkleakautovar.cpp

```diff
@@ -342,6 +342,18 @@ void CheckLeakAutoVar::checkScope(const Token * const startToken,
                     varAlloc.type = f->groupId;
                     varAlloc.status = VarInfo::ALLOC;
                 }
+                const Library::AllocFunc* g = mSettings->library.realloc(tokRightAstOperand->previous());
+                if (g && g->arg == -1) {
+                    VarInfo::AllocInfo& varAlloc = alloctype[varTok->varId()];
+                    varAlloc.type = g->groupId;
```

Comment:
shouldn't you check that the groupId matches? Write a mismatchAlloc or something if there is a mismatch.

Reply:
Yes I should. Perhaps I should also check that the pointer that should be reallocated is not already freed too?